olps/visualization/visualizer.py

```diff
@@ -1,30 +1,15 @@
 import matplotlib.pyplot as plt
 from olps.strategies.strategy import Strategy
-from typing import *
+from abc import ABC, abstractmethod
 
-class Visualizer:
+class Visualizer(ABC):
     """
-    Class for visualizing cumulative wealth of multiple strategies
+    Abstract class for visualizing portfolios over time
     """
-    outputfile: str
-    strategies: list[Strategy]
-
-    def __init__(self, outputfile: str, strategies: list[Strategy]):
-        """
-        Initialize the visualizer with already run strategies and output file
-        """
-        self.outputfile = outputfile
-        self.strategies = strategies
-
-    def visualize(self):
+    @staticmethod
+    @abstractmethod
+    def visualize(outputfile: str, strategies: list[Strategy]):
         """
         Plots the strategies to the output file in a single graph
         """
-        print(f'Attempting to write to {self.outputfile}')
-        for strategy in self.strategies:
```

Comment:
You used `self.strategies` and `self.outputfile` here, but this is a static method so `self` won't be defined. Either make this a non-static method or use the arguments provided in Line 11.

Reply:
These are lines that I deleted